### DISCUSSION TITLE
Need Help Run Java SpringBoot with OpenTelemetry cannot connect to Database

### DISCUSSION THREAD

**Discussion by moronmon** [open-telemetry/opentelemetry-java-instrumentation#5367] [Q&A]
Hello, I have a project to implement APM to OpenTelemetry and SigNoz.
Our services are running on Java SpringBoot with MongoDB, Redis, and Kafka for the support services.

I did try to run one of our services with an OpenTelemetry agent that running without connecting to any support service, and the service is running well and the metric from OpenTelemetry is well-received by SigNoz.

But I got the problem when testing the services that connect to MongoDB and Redis, the log said is the connection to both databases are failed. If I tried to run without an OpenTelemetry agent, the services can connect to both databases.

I do try to run with the debug log, but there's no luck to get any information to find out why the connection is lost to the databases.

Any one have the same issue with me?

**Comment by trask** [comment]
> But I got the problem when testing the services that connect to MongoDB and Redis, the log said is the connection to both databases are failed. If I tried to run without an OpenTelemetry agent, the services can connect to both databases.

hi @moronmon! I don't recall hearing anyone else having this issue. If you can provide more details, or ideally a standalone repro that we can use to troubleshoot, that would be great.

**Reply by moronmon** [reply]
Hi @trask thank you for the reply. I did find the problem, it seems caused by an outdated version of io.lettuce on our services.
